tcms/search/__init__.py

```diff
@@ -76,47 +74,56 @@ def _query(plan_query, run_query, case_query, target):
 
 
 def _sum_orm_queries(plans, cases, runs, target):
-    plans = plans.evaluate()
-    cases = cases.evaluate()
-    runs = runs.evaluate()
     if target == 'run':
-        if not plans and not cases:
-            if runs is None:
-                runs = TestRun.objects.none()
-        if runs is None:
-            runs = TestRun.objects.all()
-        if cases:
-            runs = runs.filter(case_run__case__in=cases).distinct()
-        if plans:
-            runs = runs.filter(plan__in=plans).distinct()
-        return runs
+        return get_run_target(plans.evaluate(), cases.evaluate(), runs.evaluate())
+
     if target == 'plan':
-        if not cases and not runs:
-            if plans is None:
-                plans = TestPlan.objects.none()
-        if plans is None:
-            plans = TestPlan.objects.all()
-        if cases:
-            plans = plans.filter(case__in=cases).distinct()
-        if runs:
-            plans = plans.filter(run__in=runs).distinct()
-        plans = plans.annotate(num_cases=Count('case', distinct=True),
-                               num_runs=Count('run', distinct=True))
-        return plans
+        return get_plan_target(plans.evaluate(), cases.evaluate(), runs.evaluate())
+
     if target == 'case':
-        if not plans and not runs:
-            if cases is None:
-                cases = TestCase.objects.none()
-        if cases is None:
-            cases = TestCase.objects.all()
-        if runs:
-            cases = cases.filter(case_run__run__in=runs).distinct()
-        if plans:
-            cases = cases.filter(plan__in=plans).distinct()
-        return cases
+        return get_case_target(plans.evaluate(), cases.evaluate(), runs.evaluate())
+
     raise ValueError('Invalid target')
 
 
+def get_run_target(plans, cases, runs):
```

Comment:
put an underscore at the start of the name of these methods, to show they are used only internally. e.g. `_get_run_target`